UnityProject/Assets/Scripts/Items/PlaceableTile.cs

```diff
@@ -116,13 +117,53 @@ public void ServerPerformInteraction(PositionalHandApply interaction)
 
```

Comment:
I like your code a lot but this method has gotten rather chonky, can you extract part of its logic into more private methods?